src/components/sideNavigation/SideNavigation.tsx

```diff
@@ -9,11 +9,26 @@ export interface ISideNavProps extends IReduxStatePossibleProps {
     opened?: boolean;
 }
 
-export const SideNavigation = (props: ISideNavProps) =>
-    <nav className={classNames(props.className, 'navigation', {'navigation-opened': props.withReduxState ? props.opened : true})}>
-        <div className='navigation-menu'>
-            <div className='navigation-menu-sections'>
-                {props.children}
-            </div>
-        </div>
-    </nav>;
+export class SideNavigation extends React.Component<ISideNavProps> {
+    static toggleEvent = 'side-navigation-toggle';
+
+    render() {
+        return (
+            <nav className={this.getClasses()}>
+                <div className='navigation-menu'>
+                    <div className='navigation-menu-sections'>
+                        {this.props.children}
+                    </div>
+                </div>
+            </nav>
+        );
+    }
+
+    private getClasses(): string {
+        return classNames(
+            this.props.className,
+            'navigation',
+            {'navigation-opened': this.props.withReduxState ? this.props.opened : true},

```

Comment:
I know that it's not your code, but this mean that if someone uses the non-connected component it will always be opened?

Reply:
yeah, i think so, we could actually just rewrite it to `{'navigation-opened': this.props.opened}`